mlflow/tensorflow/__init__.py

```diff
@@ -971,44 +979,49 @@ def _extract_n_steps(input_example_n_steps):
                             ]
 
                         else:
-                            warnings.warn(
-                                "Tensorflow keras autologging only "
-                                "supports input types of: numpy.ndarray, "
-                                "dict(<key> -> numpy.ndarray), tensorflow.data.Dataset, "
-                                "or tensorflow.keras.utils.Sequence"
+                            raise MlflowException(
+                                "Cannot log input example or model signature for input with type"
+                                f" {type(input_training_data)}. TensorFlow Keras autologging can"
+                                " only log input examples and model signatures for the following"
+                                " input types: numpy.ndarray, dict[string -> numpy.ndarray],"
+                                " tensorflow.keras.utils.Sequence, and"
+                                " tensorflow.data.Dataset (TensorFlow >= 2.1.0 required)",
+                                INVALID_PARAMETER_VALUE,
                             )
 
                         return input_example_slice
 
                     def _infer_model_signature(input_data_slice):
-                        try:
-                            original_stop_training = history.model.stop_training
-                            model_output = history.model.predict(input_data_slice)
-
-                            if (
-                                Version(tensorflow.__version__) <= Version("2.1.4")
-                                and original_stop_training
-                            ):
```

Comment:
No clear reason this needs to be scoped to a particular version. Safer to unconditionally restore the original `stop_training` value after we call `predict()`